File: app/frontend/src/core/interfaces/types/npc/EmotionSystem.ts
import { NPCData } from './npc';
import { InteractionResult } from '../../systems/npc/InteractionSystem';

export interface EmotionalState {
  primary: string;
  intensity: number;
  secondary?: string[];
  triggers: {
    type: string;
    source: string;
    impact: number;
  }[];
  duration: number;
  timestamp: number;
}

export interface EmotionSystem {
  getCurrentEmotionalState(npcId: string): Promise<string>;
  updateEmotionalState(
    npcId: string,
    trigger: string,
    intensity: number,
    context?: any
  ): Promise<EmotionalState>;
  getEmotionalResponse(
    npcId: string,
    stimulus: string,
    intensity: number
  ): Promise<{
    response: string;
    intensity: number;
  }>;
  calculateEmotionalImpact(
    baseEmotion: string,
    personality: any,
    context: any
  ): Promise<number>;
} 

File: python_converted/02_medium_complexity/src/core/interfaces/types/npc/EmotionSystem.py
from typing import Any, Dict, List



class EmotionalState:
    primary: str
    intensity: float
    secondary?: List[str]
    triggers: Dict[str, Any]
class EmotionSystem:
    getCurrentEmotionalState(npcId: str): Awaitable[str>
    updateEmotionalState(
    npcId: str,
    trigger: str,
    intensity: float,
    context?: Any
  ): Awaitable[EmotionalState>
    getEmotionalResponse(
    npcId: str,
    stimulus: str,
    intensity: float
  ): Awaitable[{
    response: str
    intensity: float>
  calculateEmotionalImpact(
    baseEmotion: str,
    personality: Any,
    context: Any
  ): Promise<number>
} 